Finish hubs

m-wynn opened the pull request:
Fixes:
* Much of #18 --- The hub location is displayed dropdown inside the new/edit Team form.  I'm hoping to leave this open and make it prettier.
* #20 --- Now all state is cleared on logout.  It will re-fetch new seasons (among other things) on login or any refresh.
* #21 --- Now there is a lot of detail on the Hub Detail page

![image](https://i.gyazo.com/170b97a8621167f6480b5c09c0418d13.png)

the-maldridge commented:
Looks really good!

In my testing I encountered a few things:
  * Wow, that pencil is kindergarten sized.  I'm confident that hub directors can hold normal sized pencils.
  * Editing the hub from the modal shows the unauthorized message, but the edit succeeds.
  * The hub menu looks good, but I'm offered the opportunity to create a new hub, something I shouldn't be allowed to do without the right capabilities.

Please fix these minor nits, then good for merge.

m-wynn commented:
* Honestly, the pencil used to be even bigger.  Looking back at it today, I realize it is still big.  Fixed :heavy_check_mark: 
* Editing the hub from the modal fails as 'unauthorized' probably because the director is trying to set himself back as the director.  Currently he does not have access to do it, but he probably should, to transfer ownership before stepping down.  I plan to optimize away redundant requests like this, but that requires reduxifying a few more things I haven't gotten to yet
* Removed that functionality from the dropdown.  If you want to add a new hub, you have to do it through the admin UI.  :heavy_check_mark:
